### PR TITLE
Lower lifetime dependence info into SIL and infer when absent 

### PR DESCRIPTION
**Pull request by meg-gupta**

### CODE REVIEW

**Comment by meg-gupta** [comment]
@swift-ci smoke test

**Comment by meg-gupta** [comment]
@swift-ci smoke test

File: test/Sema/explicit_lifetime_dependence_specifiers.swift [hunk 1]
```diff
@@ -22,7 +22,7 @@ struct WrapperStruct {
   let k: Klass
 }
 
-func invalidLifetimeDependenceOnEscapable(_ w: borrowing WrapperStruct) -> Klass {
+func invalidLifetimeDependenceOnEscapableResult(_ w: borrowing WrapperStruct) -> _borrow(w) Klass { // expected-error{{lifetime dependence specifiers can only be specified on ~Escapable results}}
```
Comment (atrick) [inline test/Sema/explicit_lifetime_dependence_specifiers.swift:25]: probably don't need to say 'specifiers' twice. "dependence can only be specified on"

**Review by atrick — APPROVED** [review]
Awesome! This is great work. I can't review all the AST implementation, but the SIL model, diagnostics, and inference look great. No need to change anything in this PR.

**Comment by meg-gupta** [comment]
@swift-ci test

**Comment by meg-gupta** [comment]
@swift-ci smoke test

**Review by xedin — COMMENTED** [review]
LGTM!

**Pull request merged**